ch_pipeline/analysis/calibration.py

```diff
@@ -2364,3 +2365,65 @@ def _calculate_uv(freq, prod, inputmap):
     uv = dist[:, np.newaxis, :] / lmbda[np.newaxis, :, np.newaxis]
 
     return uv
+
+class ExtractGaltAutoCorrelation(task.SingleTask):
+    """Extract the autocorrelations of the Galt telescope from a holography acquisition."""
+
+    def process(self, data):
+        """Extract the Galt autocorrelations and write them to disk.
+        Parameters
+        ----------
+        data: TimeStream
+            A TimeStream container holding a raw holography acquisition.
+        Returns
+        -------
+        autocorrelation: containers.GaltAutocorrelation
+            A GaltAutocorrelation container holding the extracted Galt autos
+            as a function of frequency, polarization product, and time.
+        """
+        # Redistribute over freq
+        data.redistribute("freq")
+
+        # Locate the holographic indices
+        layout_graph = layout.graph.from_db(data.time[0])
+
+        galt_inputs = get_holographic_index(
+            get_correlator_inputs(layout_graph, correlator="chime")
+        )
+
+        # Get the product map and inputs
+        prodmap = data.prod
+        ina, inb = prodmap["input_a"], prodmap["input_b"]
+
+        # Locate the Galt autocorrelations and cross-pol correlation
+        flag_YY = np.where((ina == galt_inputs[0]) & (inb == galt_inputs[0]), 1, 0)
+        flag_YX = np.where((ina == galt_inputs[0]) & (inb == galt_inputs[1]), 1, 0)
+        flag_XX = np.where((ina == galt_inputs[1]) & (inb == galt_inputs[1]), 1, 0)
+
+        auto_flag = (flag_YY + flag_YX + flag_XX).astype(bool)
+
+        # Dereference beam and weight datasets
+        beam = data.vis[:].local_array
+        weight = data.weight[:].local_array
+
+        # Load only the data corresponding to the Galt inputs
+        galt_auto = beam[:, auto_flag, :]
+        galt_weight = weight[:, auto_flag, :]
+
+        # Initialize the auto container
+        autocorrelation = containers.GaltAutocorrelation(
+            pol=np.array([b"YY", b"YX", b"XX"]),
+            attrs_from=data,
+            freq=data.freq,
+            time=data.index_map["time"],
+            comm=data.comm,
+            distributed=data.distributed,
+        )
+
```

Comment:
I'm wondering if instead of creating a separate `GaltAutocorrelation` container, we should use the existing `draco.core.containers.TimeStream` container with only 2 inputs.  I think that has everything we need except for a way to specify the polarisations.  The polarisations of the two inputs could be saved as an attribute and then for later tasks that use the autocorrelation data, you could derive the polarisation products as follows:
```
autocorrelation.attrs["pol"] = np.array([input_map[ii].pol for ii in galt_inputs])
pol_prod = [autocorrelation.attrs["pol"][aa] + autocorrelation.attrs["pol"][bb] for aa, bb in autocorrelation.prod]
``` 
The benefit of doing it this way is that we reduce the overall number of containers and we can apply other pipeline tasks that have been developed for the CHIME timestreams to the holography autocorrelation data without any modifications to those tasks.